classes/controller/menu/class-archive-import.php

```diff
@@ -0,0 +1,250 @@
+			__( 'Archive Import', 'planet4-blocks-backend' ),
+			__( 'Archive Import', 'planet4-blocks-backend' ),
+			'edit_posts',
+			'archive-import',
+			[ $this, 'get_page' ]
+		);
+	}
+
+	/**
+	 * Render the admin page.
+	 *
+	 * @return string|void The html for the admin page.
+	 */
+	public function get_page() {
+		// There is no button for this currently, needs to be manually added to the url.
+		if ( isset( $_GET['delete_all'] ) ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$archives = get_posts(
+				[
+					'post_type'   => 'archive',
+					'numberposts' => -1,
+				]
+			);
+			foreach ( $archives as $archive ) {
+				wp_delete_post( $archive->ID );
+			}
+			return 'Deleted all archived posts';
+		}
+
+		if ( isset( $_GET['done'] ) ) { //phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return 'All done';
+		}
+
+		$file = wp_import_handle_upload();
+
+		if ( isset( $file['error'] ) ) {
+			echo $file['error']; //phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+
+			wp_import_upload_form( 'edit.php?post_type=archive&page=archive-import' );
+
+			return;
+		}
+		$urls_imported_as_p4_content = self::get_imported_p3_urls();
+
+		$data = simplexml_load_string( file_get_contents( $file['file'] ) ); //phpcs:ignore WordPress.Security.NonceVerification.Recommended,WordPress.WP.AlternativeFunctions.file_get_contents_file_get_contents
+
+		$items = $data->item;
+
+		$total    = count( $items );
+		$errors   = [];
+		$existing = [];
+		$created  = [];
+
+		foreach ( $items as $item ) {
+			$url = self::WAY_BACK_URL . $item->url;
+			if ( $this->does_post_exist( $url, $urls_imported_as_p4_content ) ) {
+				$existing[] = $url;
+				continue;
+			}
+
+			$username = 'archive-user';
+
+			$author_id = $this->find_or_create_author( $username );
+
+			$result = wp_insert_post(
+				[
+					'post_author'  => $author_id,
+					'post_date'    => (string) $item->date,
+					'post_content' => (string) $item->text,
+					'post_title'   => wp_strip_all_tags( (string) $item->title ),
+					'permalink'    => $url,
+					'post_name'    => $url,
+					'guid'         => $url,
+					'post_excerpt' => (string) ( $item->excerpt ?? self::create_excerpt( $item->text ) ),
+					'post_type'    => \P4_Post_Archive::POST_TYPE,
+					'post_status'  => '1' === (string) $item->isDateMissing ? 'draft' : 'publish', // phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+				]
+			);
+
+			if ( is_plugin_active( 'sitepress-multilingual-cms/sitepress.php' ) ) {
+				global $sitepress;
+				$sitepress->set_element_language_details(
+					$result,
+					'post_archive',
+					null,
+					(string) $item->language
+				);
+			}
+
+			if ( is_wp_error( $result ) ) {
+				$errors[] = $result;
+			} else {
+				$created[] = $result;
+			}
+		}
+
+		//phpcs:disable
+		echo 'Peak memory usage: ' . $this->readable_bytes( memory_get_peak_usage() );
+		echo "Total: $total<br/>";
```

Comment:
for consistency, could you please use single quotes for string.

Reply:
This string is using interpolation so I can't replace it with single quotes.